presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -377,19 +398,113 @@ public TableStatistics getTableStatistics(ConnectorTransactionHandle transaction
+    {
+        return getColumnStatisticsValue(
+                partitionStatisticsMap,
+                column,
+                columnStatistics -> {
+                    switch (columnStatistics.getType()) {
+                        case LONG:
+                            return OptionalDouble.of(columnStatistics.getLongColumnStatistics().getNullsCount());
+                        case DOUBLE:
+                            return OptionalDouble.of(columnStatistics.getDoubleColumnStatistics().getNullsCount());
+                        case DECIMAL:
+                            return OptionalDouble.of(columnStatistics.getDecimalColumnStatistics().getNullsCount());
+                        case BOOLEAN:
+                            return OptionalDouble.of(columnStatistics.getBooleanColumnStatistics().getNullsCount());
+                        case DATE:
+                            return OptionalDouble.of(columnStatistics.getDateColumnStatistics().getNullsCount());
+                        case STRING:
+                            return OptionalDouble.of(columnStatistics.getStringColumnStatistics().getNullsCount());
+                        case BINARY:
+                            return OptionalDouble.of(columnStatistics.getBinaryColumnStatistics().getNullsCount());
+                        default:
+                            return OptionalDouble.empty();
+                    }
+                },
+                doubleStream -> {
+                    List<Double> values = doubleStream.mapToObj(Double::valueOf).collect(toList());
+                    long valuesCount = values.size();
+                    double valuesSum = values.stream().mapToDouble(Double::doubleValue).sum();
+                    if (valuesCount == 0) {
+                        return OptionalDouble.empty();
+                    }
+                    else {
+                        return OptionalDouble.of(partitionStatisticsMap.size() / valuesCount * valuesSum);
+                    }
+                });
+    }
+
+    private Estimate getColumnStatisticsValue(
+            Map<String, PartitionStatistics> partitionStatisticsMap,
+            String column,
+            Function<ColumnStatistics, OptionalDouble> valueExtractFunction,
+            Function<DoubleStream, OptionalDouble> valueAggregateFunction)
+    {
+        DoubleStream intermediateStream = partitionStatisticsMap.values().stream()
+                .map(PartitionStatistics::getColumnStatistics)
+                .filter(columnStatisticsMap -> columnStatisticsMap.containsKey(column))
+                .map(columnStatisticsMap -> columnStatisticsMap.get(column))
+                .map(valueExtractFunction)
+                .filter(OptionalDouble::isPresent)
+                .mapToDouble(OptionalDouble::getAsDouble);
+
+        OptionalDouble statisicsValue = valueAggregateFunction.apply(intermediateStream);
+
+        if (statisicsValue.isPresent()) {
```

Comment:
Can't do it here as I construct Estimate with constructor in one branch and with `.unknownValue` method in other.